### ISSUE TITLE
Add I2C support for OLED/LCD 

### ISSUE THREAD

**Issue by STemplar** [rvdbreemen/OTGW-firmware#152]
Add I2C support for LCD 16x2/20x4 and/or OLED Display with I2C Modules as suggested by the OpenTherm Gateway Manual.
![afbeelding](https://user-images.githubusercontent.com/23719607/206928743-81f8a1b0-7d00-4122-b51b-08e27121d5b0.png)

**Comment by STemplar** [comment]
Well, I am not a programmer. But if it's a standard Arduino library, I can take a look and give it a try.

**Comment by Roos-AID** [comment]
I2C is already used for temperature sensors, so not a big deal to add some function. However, it requires quite some memory for display rendering.  They key question here is what you would like to display on the LCD  and what benefit would that have over HA ?

**Comment by STemplar** [comment]
> 

As far as I know the [Arduino I2C LCD](https://arduinogetstarted.com/tutorials/arduino-lcd-i2c) interface looks very much like a serial interface. No display rendering, so perhaps a simple LCD-displey is more suited than an OLED-display.

**Comment by DaveDavenport** [comment]
u8g2 should be doable and add support for some small oled screens.
Probably nicest solution.  It is a pretty low-overhead library (I had it running in systems with 32kbyte of combined instruction and data memory). I think we have enough space remaining for this.

I can implement this if needed.

However what we should display on the display is a 2nd questions we should answer first.

**Comment by STemplar** [comment]
Indeed, I would say hardcode some variables first, like Relative Modulation level (17), Boiler flow water temperature (25) and Return water temperature (28). OEM Diagnostic code (115). Maybe later make it configurable via WebUI?

**Comment by DaveDavenport** [comment]
https://user-images.githubusercontent.com/406245/213925010-9b9dfc65-883a-41fa-a43e-5178c29d524f.mp4

**Comment by Roos-AID** [comment]
This is indeed nice add-on function, but for me it lower priority than extra sensors. How many gpio pins are extern available on the wemos ? We already need one for onewire temp sensors, one for s0 kwh counter and one is reserved for gpio output. We do need 2 more for the i2c.

**Comment by DaveDavenport** [comment]
The i2c are already exposed and you can hang a multiple sensors + oled displays on this bus.

**Comment by Roos-AID** [comment]
Dave, which IO pins are used for SDA and SCL on V2 and V2.8 boards ?  I can't find this in the manual
